pygmt/xarray/backend.py

```diff
@@ -0,0 +1,116 @@
+"""
+An xarray backend for reading raster grid/image files using the 'gmt' engine.
+"""
+
+from typing import Literal
+
+import xarray as xr
+from pygmt._typing import PathLike
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_list
+from pygmt.src.which import which
+from xarray.backends import BackendEntrypoint
+
+
+class GMTBackendEntrypoint(BackendEntrypoint):
+    """
+    Xarray backend to read raster grid/image files using 'gmt' engine.
+
+    Internally, GMT uses the netCDF C library to read netCDF files, and GDAL for GeoTIFF
+    and other raster formats. See also
+    :gmt-docs:`reference/features.html#grid-file-format`.
+
+    When using :py:func:`xarray.open_dataarray` or :py:func:`xarray.load_dataarray` with
+    ``engine="gmt"``, pass the ``raster_kind`` parameter that can be either:
+
+    - ``"grid"`` - for reading single-band raster grids
+    - ``"image"`` - for reading multi-band raster images
```

Comment:
xref: https://github.com/GenericMappingTools/pygmt/issues/3556
```suggestion
    - ``"grid"``: for reading single-band raster grids
    - ``"image"``: for reading multi-band raster images
```